### ISSUE TITLE
no-undefined-types when use infer after 54.1.0

### ISSUE THREAD

**Issue by Pavel-Boyazov** [gajus/eslint-plugin-jsdoc#1654]
## Expected behavior

No errors in console when use infer like before 54.1.0

## Actual behavior

After 54.1.0 this code shows error
% eslint .

/myfile.js
  8:1  error  The type 'I' is undefined  jsdoc/no-undefined-types
  8:1  error  The type 'I' is undefined  jsdoc/no-undefined-types

## ESLint Config

```js
import jsdoc from "eslint-plugin-jsdoc";
import globals from "globals";

export default [
	{
		plugins: { jsdoc },

		languageOptions: {
			globals: {
				...globals.node,
				NodeJS: true,
			},

			sourceType: "commonjs",
		},

		rules: {
			"jsdoc/no-undefined-types": ["error", { markVariablesAsUsed: false }],
		},
	},
];

```

## ESLint sample

```js
// Format JS code here
/**
 * @template T
 */
class Test {}

/**
 * @template {Test} T
 * @typedef {T extends Test<infer I> ? I : never} TestType
 */

console.log(Test);
```

## Environment

- Node version: v25.6.0
- ESLint version v9.39.2
- `eslint-plugin-jsdoc` version: 62.5.3
